web/modules/custom/dpl_opening_hours/dpl_opening_hours.module

```diff
@@ -77,25 +81,33 @@ function dpl_opening_hours_preprocess_page(array &$variables): void {
 }
 
 /**
- * Build the branches array.
+ * Builds an array of branch details.
  *
  * @param \Drupal\node\NodeInterface[] $branches
- *   Array of branch nodes.
+ *   An array of branch nodes.
+ *
+ * @return array<array{'branch_id': int|string|null, 'name': string|null,
+ *   'link': string, 'promoted': bool}>
+ *   An array of branch details.
  *
- * @return array<array{'branch_id': int|string|null, 'name': string|null, 'link': string, 'promoted': bool}>
- *   Array of branch details.
+ * @throws \Drupal\Core\Entity\EntityMalformedException
+ *   Thrown when the entity is malformed.
  */
 function dpl_opening_hours_build_branches_array(array $branches): array {
   $branchesArray = [];
   foreach ($branches as $branch) {
-    if ($branch instanceof NodeInterface) {
-      $branchesArray[] = [
-        'branch_id' => $branch->id(),
-        'name' => $branch->getTitle(),
-        'link' => $branch->toUrl()->toString(),
-        'promoted' => $branch->hasField('field_promoted_on_lists') && !$branch->get('field_promoted_on_lists')->isEmpty() ? (bool) $branch->get('field_promoted_on_lists')->value : FALSE,
-      ];
+    if (!($branch instanceof NodeInterface)) {
+      continue;
     }
+
+    $branchesArray[] = [
+      'branch_id' => $branch->id(),
+      'name' => $branch->getTitle(),
+      'link' => $branch->toUrl()->toString(),
+      'promoted' => $branch->hasField('field_promoted_on_lists')
+      && !$branch->get('field_promoted_on_lists')->isEmpty()
+      && (bool) $branch->get('field_promoted_on_lists')->value,
```

Comment:
This looks wrong - is it possible you're trying to do a tenary?
In php, they look like this:

```
(some_condition) ? 'true' : 'false';
```

alternatively, you can be:

```
$something = $value_if_valid ?? $fallback_value
```

Reply:
Okay, I go back to:
`'promoted' => $branch->hasField('field_promoted_on_lists') && !$branch->get('field_promoted_on_lists')->isEmpty() ? (bool) $branch->get('field_promoted_on_lists')->value : FALSE`

Reply:
Rather than long boolean statements, use temporary variables to make it more readable:

```php
$hasPromoted = $branch->hasField('field_promoted_on_lists');
$promotedNotEmpty = !$branch->get('field_promoted_on_lists')->isEmpty();
$promoted = (bool) $branch->get('field_promoted_on_lists')->value;

$branchesArray[] = [
    'branch_id' => $branch->id(),
    'name' => $branch->getTitle(),
    'link' => $branch->toUrl()->toString(),
    'promoted' => $hasPromoted && $promotedNotEmpty ? $promoted : FALSE,
];
```

But we can simplify it. `$branch->get('field_promoted_on_lists')->value` returns null if it's empty, which is the same as false when casted to bool:

```php
$hasPromoted = $branch->hasField('field_promoted_on_lists');
$promoted = (bool) $branch->get('field_promoted_on_lists')->value;

$branchesArray[] = [
    'branch_id' => $branch->id(),
    'name' => $branch->getTitle(),
    'link' => $branch->toUrl()->toString(),
    'promoted' => $hasPromoted && $promoted,
];
```

But... We know that branches has the field, it's written to the configuration, and if someone deletes it, we'd rather have thing explode. So:

```php
$branchesArray[] = [
    'branch_id' => $branch->id(),
    'name' => $branch->getTitle(),
    'link' => $branch->toUrl()->toString(),
    'promoted' => (bool) $branch->get('field_promoted_on_lists')->value,
];
```

Reply:
Thanks for the good explanation. I have corrected it  🙏